test/controllers/api/v2/api_controller_test.rb

```diff
@@ -21,6 +21,20 @@ def teardown
       @options = nil
     end
 
+    def test_check_katello_agent_not_disabled
+      ::Katello.expects(:with_katello_agent?).returns(true)
+      @controller.expects(:fail).never
+
+      @controller.check_katello_agent_not_disabled
+    end
+
+    def test_check_katello_agent_not_disabled_fail
+      ::Katello.expects(:with_katello_agent?).returns(false)
+      @controller.expects(:fail).with(HttpErrors::BadRequest, "This action uses katello-agent, which is currently disabled. Use remote execution instead.")
```

Comment:
You can write this test such that it is less coupled to the implementation, ex:

```
error = assert_raises(HttpErrors::BadRequest) do
  @controller.check_katello_agent_not_disabled
end
```

and if you want to validate the message:

```
assert_matches(/remote execution instead/, error.message)
```

Reply:
Not asking to change it, just sharing.